### ISSUE TITLE
Assertion fails when assigning a class X to the type typeOf(X)

### ISSUE THREAD

**Issue by fabiandev** [codemix/flow-runtime#84]
This is a:

 - [x] Bug Report
 - [ ] Feature Request
 - [x] Question
 - [ ] Other

Which concerns:

 - [x] flow-runtime
 - [ ] babel-plugin-flow-runtime
 - [ ] flow-runtime-validators
 - [ ] flow-runtime-mobx
 - [ ] flow-config-parser
 - [ ] The documentation website

---

### What is the current behaviour?

Below are the code snippets of a class in flow and in JavaScript with flow-runtime type annotations.

```ts
class X {
  static prop1: number = 1;
  prop2: string;
  static method(a: string): number { return 1; }
}
```

```js
import t from "flow-runtime";
@t.annotate(t.class("X", t.staticProperty("prop1", t.number()), t.property("prop2", t.string()), t.staticMethod("method", t.param("a", t.string()), t.return(t.number()))))
class X {
  @t.decorate(t.number())
  static prop1 = 1;

  static method(a) {
    let _aType = t.string();

    const _returnType = t.return(t.number());

    t.param("a", _aType).assert(a);
    return _returnType.assert(1);
  }
}
```

When performing the following assignment

```ts
let x: typeof X = X;
```

```js
let _xType = t.typeOf(X), x = _xType.assert(X);
```

the assertion fails with the error below:

```
 X must be an object

Expected: {
  static prop1: number;
  prop2: string;
  static method: (a: string) => number;
}

Actual: X
```

I even tried assigning typeof X:

```js
let _xType = _t.typeOf(X), x = _xType.assert(t.typeOf(X));
```

which resulted in the following error:

```
 X.constructor.prop1 must be a number

Expected: number

Actual: void

-------------------------------------------------

X.prop2 must be a string

Expected: string

Actual: void

-------------------------------------------------

X.constructor.method must be a function

Expected: (a: string) => number

Actual: void
```

---

### What is the expected behaviour?

Please correct me if I'm wrong, but I think at least ```let x: typeof X = X``` shouldn't error, as `X` actually holds the type representation for the class, so ```typeOf(X)``` should be comparable to ```X```.

---

### Which package versions are you using?

I used the live demo: https://codemix.github.io/flow-runtime

**Comment by fabiandev** [comment]
It seems when annotating a class with a separate variable, it is possible to assign the type as intended.

```js
var TestClass = @t.annotate(t.class("TestClass"))
class TestClass {};

let _xType = t.typeOf(TestClass);
```

With the declaration above, the following assertions do not error:

```js
let x = _xType.assert(t.ref(TestClass));
```

```js
let x = _xType.assert(t.typeOf(TestClass));
```

**Comment by phpnode** [comment]
fixed in 0.11.1
